common/tools/dev-tool/src/util/customization/classes.ts

```diff
@@ -0,0 +1,357 @@
+  }
+}
+
+export function augmentConstructor(
+  customConstructor: ConstructorDeclaration,
+  originalConstructor: ConstructorDeclaration | undefined,
+  originalClassDeclaration: ClassDeclaration
+) {
+  const constructorComments = getComments(customConstructor, originalConstructor);
+  addConstructorsToClass(customConstructor, originalClassDeclaration, constructorComments);
+}
+
+export function augmentMethod(
+  originalMethod: MethodDeclaration | undefined,
+  customMethod: MethodDeclaration,
+  originalClass: ClassDeclaration
+) {
+  const methodComments = getComments(customMethod, originalMethod);
+  // custom is adding a new method this is a new method on the class, we'll add it to original
+  if (!originalMethod) {
+    addMethodToClass(customMethod, originalClass, methodComments);
+    return;
+  }
+
+  // Original method with the same name exists so
+  // we need to check if the custom method is using the original method
+  // to determine if we need to augment or replace
+  if (isAugmentingMethod(customMethod)) {
+    convertToPrivateMethod(originalMethod, originalClass);
+    addMethodToClass(customMethod, originalClass, methodComments);
+  } else {
+    // This is not using the original method so we'll replace it
+    originalMethod.remove();
+    addMethodToClass(customMethod, originalClass, methodComments);
+  }
+}
+
+export function isAugmentingConstructor(
+  customConstructor: ConstructorDeclaration,
+  originalConstructor?: ConstructorDeclaration
+): boolean {
+  const customConstructorContent = customConstructor.getBody()?.getFullText();
+
+  if (!originalConstructor) {
+    // If there is no original constructor, we'll just copy the custom one
+    return false;
+  }
+
+  if (customConstructorContent?.includes(`@azsdk-constructor`)) {
+    return true;
+  }
+
+  return false;
+}
+
+export function getConstructorAugmentationParameters(
+  customConstructor: ConstructorDeclaration,
+  originalParameters?: ParameterDeclaration[]
+): Map<string, string> {
+  const expectedAugmentationParams = new Map<string, string>();
+
+  if (!originalParameters) {
+    return expectedAugmentationParams;
+  }
+
+  for (const originalParameter of originalParameters) {
+    expectedAugmentationParams.set(
+      originalParameter.getName(),
+      `${AUGMENT_CLASS_TOKEN}${originalParameter.getName()}`
+    );
+  }
+
+  for (const [_, augmentedParameter] of expectedAugmentationParams) {
+    if (!customConstructor.getBodyText()?.includes(augmentedParameter)) {
+      throw new Error(`Custom constructor is missing parameter ${augmentedParameter}`);
+    }
+  }
+
+  return expectedAugmentationParams;
+}
+
+export function isAugmentingMethod(customMethod: MethodDeclaration): boolean {
+  const customMethodContent = customMethod.getBody()?.getFullText();
+
+  if (customMethodContent?.includes(`this.${AUGMENT_CLASS_TOKEN}`)) {
+    return true;
+  }
+
+  return false;
```

Comment:
Could this be simplified to
```suggestion
  return customMethod.getBody()?.getFullText()?.includes(`this.${AUGMENT_CLASS_TOKEN}`)
```